File: app/src/main/java/org/wikipedia/feed/model/ListCard.java
package org.wikipedia.feed.model;

import android.support.annotation.NonNull;

import org.wikipedia.dataclient.WikiSite;

import java.util.Collections;
import java.util.List;

public abstract class ListCard<T extends Card> extends WikiSiteCard {
    @NonNull private final List<T> items;

    public ListCard(@NonNull List<T> items, @NonNull WikiSite wiki) {
        super(wiki);
        this.items = Collections.unmodifiableList(items);
    }

    @NonNull public List<T> items() {
        return items;
    }
}

File: app/src/main/java/org/wikipedia/feed/model/ListCard.kt
package org.wikipedia.feed.model

import org.wikipedia.dataclient.WikiSite
import java.util.*

abstract class ListCard<T : Card>(internal val items: List<T>,
                                  wiki: WikiSite) : WikiSiteCard(wiki) {

    fun items(): List<T> {
        return Collections.unmodifiableList(items)
    }
}
